warehouse/query-core/src/main/java/datawave/query/iterator/QueryLogIterator.java

```diff
@@ -0,0 +1,164 @@
+package datawave.query.iterator;
+
+import org.apache.accumulo.core.data.ByteSequence;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Range;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.IteratorEnvironment;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+import org.apache.accumulo.core.iterators.YieldCallback;
+import org.apache.log4j.Logger;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Map;
+
+import static datawave.query.iterator.QueryOptions.QUERY_ID;
+
+public class QueryLogIterator implements SortedKeyValueIterator<Key, Value>{
+
+    protected static final Logger log = Logger.getLogger(QueryLogIterator.class);
+
+    protected SortedKeyValueIterator<Key,Value> source;
+    protected SortedKeyValueIterator<Key,Value> sourceForDeepCopies;
```

Comment:
sourceForDeepCopies not needed, not used.

Reply:
What we should be doing is attempting to pull the query id from the queryOptions for use in log messages.

Reply:
That would be better than stoiring the entire map of options.